### ISSUE TITLE
[BUG]: Output of step raises error in mypy

### ISSUE THREAD

**Issue by duarteocarmo** [zenml-io/zenml#1491]
### Contact Details [Optional]

duarteocarmo@gmail.com

### System Information

ZENML_LOCAL_VERSION: 0.38.0
ZENML_SERVER_VERSION: 0.38.0
ZENML_SERVER_DATABASE: sqlite
ZENML_SERVER_DEPLOYMENT_TYPE: local
ZENML_CONFIG_DIR: /Users/duartecarmo/Library/Application Support/zenml
ZENML_LOCAL_STORE_DIR: /Users/duartecarmo/Library/Application Support/zenml/local_stores
ZENML_SERVER_URL: http://127.0.0.1:8237
ZENML_ACTIVE_REPOSITORY_ROOT: None
PYTHON_VERSION: 3.10.10
ENVIRONMENT: native
SYSTEM_INFO: {'os': 'mac', 'mac_version': '13.2.1'}
ACTIVE_WORKSPACE: default
ACTIVE_STACK: default
ACTIVE_USER: default
TELEMETRY_STATUS: enabled
ANALYTICS_CLIENT_ID: 53a69fdb-e650-414d-932a-93da77065021
ANALYTICS_USER_ID: e1626071-62f7-4fb3-92a8-2bbb93aad4ca
ANALYTICS_SERVER_ID: 53a69fdb-e650-414d-932a-93da77065021
INTEGRATIONS: ['graphviz', 'kaniko', 'lightgbm', 'pillow', 'plotly', 'scipy', 'sklearn', 'xgboost']
PACKAGES: {'xyzservices': '2023.2.0', 'certifi': '2022.12.7', 'pytz': '2022.7.1', 'setuptools': '65.5.0', 'cryptography': '40.0.1', 'pyzmq': '25.0.0', 'ocmodel': '23.2.17', 'black': '23.1.0', 'pip': '23.0.1', 'packaging': '23.0', 'attrs': '22.2.0', 'aiofiles': '22.1.0', 'argon2-cffi': '21.3.0', 'argon2-cffi-bindings':
'21.2.0', 'isoduration': '20.11.0', 'rich': '12.6.0', 'websockets': '11.0.1', 'pyarrow': '11.0.0', 'pillow': '9.4.0', 'ipython': '8.11.0', 'tenacity': '8.2.2', 'click': '8.1.3', 'jupyter-client': '8.0.3', 'ipywidgets': '7.7.5', 'nbconvert': '7.2.9', 'pytest': '7.2.2', 'coverage': '7.2.1', 'python-box': '7.0.1', 'ipykernel':
'6.21.3', 'notebook': '6.5.3', 'tornado': '6.2', 'docker': '6.0.1', 'bleach': '6.0.0', 'plotly': '5.13.1', 'psutil': '5.9.4', 'traitlets': '5.9.0', 'nbformat': '5.7.3', 'pyyaml': '5.4.1', 'cachetools': '5.3.0', 'jupyter-core': '5.2.0', 'decorator': '5.1.1', 'tokenize-rt': '5.0.0', 'smmap': '5.0.0', 'tqdm': '4.65.0',
'fonttools': '4.39.0', 'protobuf': '4.22.1', 'jsonschema': '4.17.3', 'beautifulsoup4': '4.11.2', 'rsa': '4.9', 'pexpect': '4.8.0', 'gdown': '4.7.1', 'typing-extensions': '4.5.0', 'uritemplate': '4.1.1', 'gitdb': '4.0.10', 'bcrypt': '4.0.1', 'pytest-cov': '4.0.0', 'filelock': '3.10.7', 'orjson': '3.8.10', 'matplotlib':
'3.7.1', 'widgetsnbextension': '3.6.4', 'anyio': '3.6.2', 'jupyterlab': '3.6.1', 'asgiref': '3.6.0', 'pyproj': '3.4.1', 'idna': '3.4', 'lightgbm': '3.3.5', 'python-jose': '3.3.0', 'gitpython': '3.1.31', 'jinja2': '3.1.2', 'platformdirs': '3.1.0', 'charset-normalizer': '3.1.0', 'threadpoolctl': '3.1.0', 'prompt-toolkit':
'3.0.38', 'geojson': '3.0.1', 'networkx': '3.0', 'google-api-python-client': '2.80.0', 'requests': '2.28.2', 'pycparser': '2.21', 'jupyterlab-server': '2.20.0', 'fastjsonschema': '2.16.3', 'google-auth': '2.16.2', 'pygments': '2.14.0', 'babel': '2.12.1', 'google-api-core': '2.11.0', 'python-dateutil': '2.8.2',
'google-cloud-storage': '2.7.0', 'mapclassify': '2.5.0', 'munch': '2.5.0', 'pyparsing': '2.4.7', 'google-resumable-media': '2.4.1', 'jupyter-server': '2.4.0', 'soupsieve': '2.4', 'google-cloud-core': '2.3.2', 'whitebox': '2.3.1', 'watchdog': '2.3.1', 'pyshp': '2.3.1', 'jsonpointer': '2.3', 'whiteboxgui': '2.3.0', 'asttokens':
'2.2.1', 'cloudpickle': '2.2.1', 'markupsafe': '2.1.2', 'python-json-logger': '2.0.7', 'mistune': '2.0.5', 'shapely': '2.0.1', 'tomli': '2.0.1', 'ipyevents': '2.0.1', 'iniconfig': '2.0.0', 'googleapis-common-protos': '1.58.0', 'geocoder': '1.38.1', 'urllib3': '1.26.14', 'numpy': '1.23.5', 'six': '1.16.0', 'cffi': '1.15.1',
'webcolors': '1.12', 'pydantic': '1.10.6', 'scipy': '1.10.1', 'backoff': '1.10.0', 'fiona': '1.9.1', 'pyperclip': '1.8.2', 'alembic': '1.8.1', 'send2trash': '1.8.0', 'distro': '1.8.0', 'passlib': '1.7.4', 'xgboost': '1.7.4', 'pysocks': '1.7.1', 'logzero': '1.7.0', 'debugpy': '1.6.6', 'monotonic': '1.6', 'nest-asyncio':
'1.5.6', 'pandas': '1.5.3', 'fqdn': '1.5.1', 'websocket-client': '1.5.1', 'google-crc32c': '1.5.0', 'rfc3986': '1.5.0', 'pandocfilters': '1.5.0', 'sqlalchemy': '1.4.41', 'kiwisolver': '1.4.4', 'analytics-python': '1.4.post1', 'sniffio': '1.3.0', 'mako': '1.2.4', 'arrow': '1.2.3', 'scikit-learn': '1.2.2', 'tinycss2': '1.2.1',
'executing': '1.2.0', 'joblib': '1.2.0', 'uri-template': '1.2.0', 'jupyterlab-widgets': '1.1.4', 'catboost': '1.1.1', 'click-plugins': '1.1.1', 'exceptiongroup': '1.1.0', 'contourpy': '1.0.7', 'pymysql': '1.0.3', 'pycrs': '1.0.2', 'mypy-extensions': '1.0.0', 'pluggy': '1.0.0', 'python-dotenv': '1.0.0', 'fastapi': '0.75.2',
'numba': '0.56.4', 'shap': '0.41.0', 'wheel': '0.40.0', 'llvmlite': '0.39.1', 'sqlalchemy-utils': '0.38.3', 'zenml': '0.38.0', 'httpx': '0.23.3', 'geemap': '0.20.5', 'graphviz': '0.20.1', 'pyrsistent': '0.19.3', 'httplib2': '0.19.1', 'future': '0.18.3', 'validators': '0.18.2', 'jedi': '0.18.2', 'ecdsa': '0.18.0', 'aiosqlite':
'0.18.0', 'uvicorn': '0.17.6', 'ipyleaflet': '0.17.2', 'terminado': '0.17.1', 'starlette': '0.17.1', 'uvloop': '0.17.0', 'httpcore': '0.16.3', 'prometheus-client': '0.16.0', 'h11': '0.14.0', 'folium': '0.14.0', 'bqplot': '0.12.37', 'geopandas': '0.12.2', 'seaborn': '0.12.2', 'pathspec': '0.11.0', 'cycler': '0.11.0',
'python-terraform': '0.10.1', 'json5': '0.9.11', 'commonmark': '0.9.1', 'parso': '0.8.3', 'ypy-websocket': '0.8.2', 'watchgod': '0.8.2', 'jupyter-server-fileid': '0.8.0', 'pickleshare': '0.7.5', 'nbclient': '0.7.2', 'cligj': '0.7.2', 'defusedxml': '0.7.1', 'scooby': '0.7.1', 'ptyprocess': '0.7.0', 'jupyter-events': '0.6.3',
'stack-data': '0.6.2', 'jupyter-server-ydoc': '0.6.1', 'ipyfilechooser': '0.6.0', 'branca': '0.6.0', 'y-py': '0.5.9', 'geeadd': '0.5.6', 'nbclassic': '0.5.3', 'webencodings': '0.5.1', 'httptools': '0.5.0', 'pyasn1': '0.4.8', 'jupyter-server-terminals': '0.4.4', 'jupyter-black': '0.3.3', 'click-params': '0.3.0',
'pyasn1-modules': '0.2.8', 'wcwidth': '0.2.6', 'sankee': '0.2.3', 'jupyter-ydoc': '0.2.2', 'ipytree': '0.2.2', 'pure-eval': '0.2.2', 'jupyterlab-pygments': '0.2.2', 'notebook-shim': '0.2.2', 'fastapi-utils': '0.2.1', 'traittypes': '0.2.1', 'backcall': '0.2.0', 'ipython-genutils': '0.2.0', 'ffmpeg-python': '0.2.0',
'earthengine-api': '0.1.349', 'pygwalker': '0.1.6.1', 'ratelim': '0.1.6', 'matplotlib-inline': '0.1.6', 'colour': '0.1.5', 'rfc3339-validator': '0.1.4', 'appnope': '0.1.3', 'comm': '0.1.2', 'rfc3986-validator': '0.1.1', 'google-auth-httplib2': '0.1.0', 'ruff': '0.0.254', 'ee-extra': '0.0.15', 'sqlmodel': '0.0.8', 'slicer':
'0.0.7', 'python-multipart': '0.0.6', 'eerepr': '0.0.4', 'sqlalchemy2-stubs': '0.0.2a33'}

CURRENT STACK

Name: default
ID: 91937593-9730-4852-a816-773a10cda15a
Shared: No
User: default / e1626071-62f7-4fb3-92a8-2bbb93aad4ca
Workspace: default / e5125738-3312-4854-9f7a-5e44908d7945

ORCHESTRATOR: default

Name: default
ID: e31ff7ff-3031-4858-8431-69f322ff290b
Type: orchestrator
Flavor: local
Configuration: {}
Shared: No
User: default / e1626071-62f7-4fb3-92a8-2bbb93aad4ca
Workspace: default / e5125738-3312-4854-9f7a-5e44908d7945

ARTIFACT_STORE: default

Name: default
ID: 6260763a-a9b1-4ed4-8a06-246c7d4a6d71
Type: artifact_store
Flavor: local
Configuration: {'path': ''}
Shared: No
User: default / e1626071-62f7-4fb3-92a8-2bbb93aad4ca
Workspace: default / e5125738-3312-4854-9f7a-5e44908d7945

### What happened?

![image](https://user-images.githubusercontent.com/26342344/232024824-26ccf76f-fbaa-4988-944c-a3d2119358f8.png)

### Reproduction steps

1.
2.
3.
...

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by strickvl** [comment]
Just wanted to confirm, since it's not shown in the snippet you pasted in here, that you're importing `Output` using `from zenml.steps import Output`?

**Comment by duarteocarmo** [comment]
Yup, here's a more compreheensive snippet: 
```python
from zenml.steps import BaseParameters, Output, step


class TrainingDataParams(BaseParameters):
    data_location: str
    apply_log_to_target: bool = True
    random_seed: int = 42
    test_size: float = 0.2


@step
def load_training_data(params: TrainingDataParams) -> pandas.DataFrame:
    dataframe = pandas.read_csv(params.data_location)
    return dataframe


@step
def prepare_features(
    dataframe: pandas.DataFrame, params: TrainingDataParams
) -> Output(X=pandas.DataFrame, y=pandas.Series):
    # remove unnamed
    to_drop = [col for col in dataframe.columns if "Unnamed" in col]
    dataframe = dataframe.drop(columns=to_drop)
    # ...
```
